pkg/consensus/client.go

```diff
@@ -49,20 +52,29 @@ func NewClient(ctx context.Context, endpoint string, logger *zap.Logger, current
 		Codec: eth2api.JSONCodec{},
 	}
 
+	proposerCache, err := lru.New(cacheSize)
+	if err != nil {
+		return &Client{}, err
+	}
+	executionCache, err := lru.New(cacheSize)
+	if err != nil {
+		return &Client{}, err
```

Comment:
```suggestion
		return nil, err
```

I think we can just return `nil` here